pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -417,25 +417,39 @@ func (m *kubeGenericRuntimeManager) podSandboxChanged(pod *v1.Pod, podStatus *ku
 		}
 	}
 
-	// Needs to create a new sandbox when readySandboxCount > 1 or the ready sandbox is not the latest one.
+	//check the pod is succeeded
+	succeededContainerCount := 0
+	for _, container := range pod.Spec.Containers {
+		for _, status := range podStatus.ContainerStatuses {
+			if container.Name == status.Name {
+				if status.State == kubecontainer.ContainerStateExited && status.ExitCode == 0 {
```

Comment:
I think this depends on the restart policy? If the restart policy is `Always`, the sandbox should still be recreated. If the restart policy is `Never`, then failed containers should be included.

Reply:
Yes.
Now the code show us that:
when the restart policy is not  Always and the all container is succeded, the sandbox will not create a new one.
Others will enter the old process.